File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int L = 0;
    int W = 1;
    int S = 0;
    //prompt for text
    string text = get_string("Text: ");
    
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //find numbers of letters
        if (isalpha(text[i]))
        {
            L++;
        }
        //find numbers of words
        else if (text[i] == ' ')
        {
            W++;
        }
        //find numbers of sentences
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            S++;
        }
    }

    //cal index
    float index = 0.0588 * L / W * 100 - 0.296 * S / W * 100 - 15.8;
    //round to int
    int grade = round(index);
    //print outcome
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: readability.py
from cs50 import get_string

# Get user text
text = get_string("Text: ")

# Calculate letters, words and sentences in text
lttr = 0
words = 1
stc = 0
for c in text:
    # Cal letters
    if c.isalpha():
        lttr += 1
    # Cal words    
    elif c == " ":
        words += 1
    # Cal sentences
    elif c == "." or c == "?" or c == "!":
        stc += 1

# Cal index
index = round(0.0588 * lttr / words * 100 - 0.296 * stc / words * 100 - 15.8)
    
# Print results
if index > 15:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

